docs/getting-started/0-agent.md

```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
+
+After creating the Agent object, we need to create and set the Inbound/Outbound transports that will handle traffic to/from the Agent.
+
+
+```ts
+  const httpOutboundTransporter = new HttpOutboundTransporter();
+  agent.setOutboundTransporter(httpOutboundTransporter);
+
+  //Inbound transports are currently built-in, you don't need to create them.
+```
+
+More about [Transports](1-transports.md).
+
+
+### 6- Init Agent
+
+After creating the Agent object and configuring it, we initialize the Agent.
+
+```ts
+
+//Its highly recommended to wrap the initialization flow in a try/catch block 
+try
+{
+  await agent.initialize();
+  console.log('Initialized agent!');
+}catch(e){
+  console.log(`Agent init error:${e}`);
 }
+```
+
 
-const agent = new Agent(agentConfig)
+### 6- Handling State Changes
+
+After you successfully initialize your Agent, you will notice that most of the hard work is being done by the underlying Aries/Indy framework. However, you as a controller will need to intercept in some situations when there is a state change (like when you need to accept an offered credential or reject/accept an incoming connection request). This is done through state change handlers.
+
+#### Creating Event(State) Handlers 
+
+```ts
+
+agent.events.on(
+      AgentEventTypes.AgentMessageReceived,
+      handleBasicMessageReceive,
+);
+
+agent.events.on<CredentialStateChangedEvent>(
+  CredentialEventTypes.CredentialStateChanged,
+  event => {
+    handleCredentialStateChange(event); //custom callback for handling any state change when offering/receiving VCs
+  },
+);
+
+agent.events.on<ConnectionStateChangedEvent>(
+  ConnectionEventTypes.ConnectionStateChanged,
+  event => {
+    handleConnectionStateChange(event); //custom callback for handling any state change when offering/receiving a connection
+  },
+);
+```
+
+Example: This sample credential callback shows how to detect that a credential is received, show the user the credential asserts and give the option to accept/reject the offered credential.
+
+```ts
+const handleCredentialStateChange = async (
+  event: CredentialStateChangedEvent,
+) => {
+  console.log(
+    `>> Credential state changed: ${event.payload.credentialRecord.id}, previous state -> ${event.payload.previousState} new state: ${event.payload.credentialRecord.state}`,
+  );
+
+  if (event.payload.credentialRecord.state === 'offer-received') {
+    console.log(`>> Received offer, should display credential to user`);
+    
+    //Display offer to user 
+    //..
+    //On user click "Accept"
+    console.log(`>>ACCEPTING OFFER`);
+    agent.credentials.acceptOffer(event.payload.credentialRecord.id);
+
+  } else if (event.payload.credentialRecord.state === 'credential-received') {
+
+    //Credential recieved and stored in the wallet
+    console.log('>> Received Credential');
+    await agent.credentials.acceptCredential(event.payload.credentialRecord.id);
+    console.log('ALL DONE - CREDENTAIL ACCEPTED');
+  }
```

Comment:
With content approved mode we can remove this part. 

Maybe adding a `CredentialState.Done` listener to log when the process completed will be helpful